Packs/ApiModules/Scripts/OktaApiModule/OktaApiModule.py

```diff
@@ -0,0 +1,252 @@
+
+        return token
+
+    def initial_setup(self):
+        """
+        Initial setup for the first time the integration is used.
+        """
+        integration_context = get_integration_context()
+
+        if integration_context.get('initialized', False):  # If the initial setup was already done, do nothing
+            return
+
+        if self.auth_type == AuthType.OAUTH:
+            # Add "SUPER_ADMIN" role to client application, which is required for OAuth authentication
+            try:
+                self.assign_app_role(client_id=self.client_id, role="SUPER_ADMIN", auth_type=AuthType.API_TOKEN)
+                demisto.debug("'SUPER_ADMIN' role has been assigned to the client application.")
+
+            except DemistoException as e:
+                # If the client application already has the "SUPER_ADMIN" role, ignore the error.
+                # E0000090 Error code official docs description: Duplicate role assignment exception.
+                if e.res.headers.get('content-type') == 'application/json' and e.res.json().get('errorCode') == 'E0000090':
+                    demisto.debug('The client application already has the "SUPER_ADMIN" role assigned.')
+
+                else:
+                    raise e
+
+            self.get_token()
+
+        integration_context = get_integration_context()
+        integration_context['initialized'] = True
+        set_integration_context(integration_context)
+
+    def _http_request(self, method, url_suffix='', full_url=None, headers=None, auth=None, json_data=None,
+                      params=None, data=None, files=None, timeout=None, resp_type='json', ok_codes=None,
+                      return_empty_response=False, retries=0, status_list_to_retry=None,
+                      backoff_factor=5, raise_on_redirect=False, raise_on_status=False,
+                      error_handler=None, empty_valid_codes=None, params_parser=None,
+                      auth_type: AuthType | None = None, **kwargs):
+        """
+        Override BaseClient._http_request() to automatically add authentication headers.
+
+        Args:
+            auth_type (AuthType | None): Type of authentication to use for the request.
+                If not provided, 'self.use_oauth' will be used.
+        """
+        auth_type = auth_type if auth_type is not None else self.auth_type
+        auth_headers = {}
+
+        if auth_type == AuthType.OAUTH:
+            auth_headers['Authorization'] = f'Bearer {self.get_token()}'
+
+        elif auth_type == AuthType.API_TOKEN:
+            auth_headers['Authorization'] = f'SSWS {self.api_token}'
+
+        original_headers = kwargs.get('headers', {})
+        kwargs['headers'] = {**auth_headers, **original_headers}
+        return super()._http_request(**kwargs)
+
+
+def reset_integration_context():
+    """
+    Reset the integration context.
+    """
+    integration_context = get_integration_context()
+    integration_context["token"] = "XXX"
```

Comment:
why this is considered as 'reset'
if get_token will called within the time period - it will return 'XXX' instead of generate new one, no?

Reply:
The `XXX` replacement is just to avoid the token being printed to logs.

The actual reset is by calling `set_integration_context({})` on the next line.
Before resetting the integration context, we store it in a variable, replace the token (just to censor it for logs), and then print the integration context as it was before it was reset (line 252).

As for why we log this - I just saw we do on Microsoft integrations, and was told that having the integration context logged before resetting helped debugging in some cases, so I thought I'll log it here as well.

Reply:
i missed it ..